### PR TITLE
Expand audits when score is not perfect

### LINKED ISSUE #1754: Report: aggregations are marked as passing even if one fails
screen shot 2017-02-19 at 3 42 59 pm

### PR DESCRIPTION
**Pull request by ebidel**
R: all

Fixes #1754

### CODE REVIEW

File: lighthouse-core/report/handlebar-helpers.js [hunk 1]
```diff
@@ -63,10 +63,9 @@ const handlebarHelpers = {
     return calculateRating(totalScore);
   },
 
-  // Converts an aggregation's score to a rating used for styling.
+  // Converts an aggregation's score to a pass/fail rating.
   getAggregationScoreRating: score => {
-    const rating = calculateRating(score * 100);
-    return rating !== 'good' ? 'poor' : rating; // avg rating maps to failure for aggregation.
+    return score === 1 ? 'good' : 'poor';
```
Comment (paulirish) [inline lighthouse-core/report/handlebar-helpers.js]: this doesn't affect our scored aggregations like "page load performance is fast"? I wouldn't expect a binary score for that one.

Comment (ebidel) [inline lighthouse-core/report/handlebar-helpers.js]: This is strictly for the collapsing behavior. Doesn't affect the score in any way. For "page load performance is fast", the aggregation will close if everything is 100/100 and open if 1-more audits is not 100/100. So if one of the audits is 98/100 and everything else is 100%, we'll still show you were you can do a bit better.

Comment (ebidel) [inline lighthouse-core/report/handlebar-helpers.js]: Pics!

All audits are 100/100:

screen shot 2017-02-21 at 1 35 16 pm

Some audits are not 100/100:

screen shot 2017-02-21 at 1 35 11 pm

Comment (ebidel) [inline lighthouse-core/report/handlebar-helpers.js]: Haha. Is the 2nd screenshot confusing? (all audits green but the overall is a red x)

Comment (paulirish) [inline lighthouse-core/report/handlebar-helpers.js]: yes. :)  I have a feeling that `calculateRating` is returning ~0.9 in that case.. thus the question. :)

Comment (ebidel) [inline lighthouse-core/report/handlebar-helpers.js]: What do you think all boolean audits should do?

screen shot 2017-02-21 at 2 11 47 pm

IMO, the overall icon should be orange even though the score is technically green 5/6 => 83/100.

Comment (patrickhulce) [inline lighthouse-core/report/handlebar-helpers.js]: Yeah the second one was very confusing for me as well, haha. I'm on board with green for the first one but still expanded and with the orange for the boolean audits.

Comment (paulirish) [inline lighthouse-core/report/handlebar-helpers.js]: sgtm. 

i like orange icon for the group _(Avoiding APIs…)_ I'm thinking the icon being circle with horizontal line through it?

Comment (ebidel) [inline lighthouse-core/report/handlebar-helpers.js]: Horizontal lines we're using for errors. Orange X would be ok. "Here's something to look at, but this pretty means you've done pretty good."

Comment (paulirish) [inline lighthouse-core/report/handlebar-helpers.js]: wfm

Comment (ebidel) [inline lighthouse-core/report/handlebar-helpers.js]: done

**Comment by ebidel** [comment]
This should be doing the RightThing™ now. Some examples:

screen shot 2017-02-24 at 10 25 48 pm
screen shot 2017-02-24 at 10 26 20 pm
screen shot 2017-02-24 at 10 26 31 pm

**Comment by ebidel** [comment]
Also added the aggregation score to the DOM so it's easier to debug for users.

**Comment by robdodson** [comment]
Hm orange X versus red X creates an a11y issue because you're using color as the only modality for conveying that information. If it's possible to add an additional icon like a `-` for orange elements that would help

**Comment by ebidel** [comment]
One though: It might be ok if the orange X is interpreted as red OR orange for these cases. Scored audits contain numbers conveying info but boolean audits just open the dropdowns.

Alternatively, we could do an orange ✔️ or "!".

**Comment by robdodson** [comment]
+1 for !

**Pull request merged**